crates/ty_python_semantic/src/types/visitor.rs

```diff
@@ -236,6 +238,9 @@ fn walk_non_atomic_type<'db, V: TypeVisitor<'db> + ?Sized>(
             visitor.visit_protocol_instance_type(db, protocol);
         }
         NonAtomicType::TypedDict(typed_dict) => visitor.visit_typed_dict_type(db, typed_dict),
+        NonAtomicType::TypeAlias(alias) => {
+            visitor.visit_type(db, alias.value_type(db));
+        }
```

Comment:
Should we add a `visit_type_alias_type` to the trait so that concrete implementations of the trait have it available as a hook?